### DISCUSSION TITLE
Do I mark a function as Changed so that its reevaluated and not cached in Rete Algorithm

### DISCUSSION THREAD

**Discussion by shreyaspurohit** [hyperjumptech/grule-rule-engine#301] [Q&A] [answered]
Should I be marking a function to be reevaluated if I know the underlying was changed by a rule execution so that it is reevaluated in the next cycle?

For example `Changed("ResultFact.MapHasResult(\"x\")")`

Is that the correct syntax?

Thank you for any pointers.

**Comment by newm4n** [comment, marked as answer]
Hi @shreyaspurohit  , 
Yes, that's how you do it.

Looking at the logic of `Reset` function in `WorkingMemory.go`

```go
// Reset will reset the evaluated status of a specific variable if its contains a variable name in its signature.
// Returns true if any expression was reset, false if otherwise
func (e *WorkingMemory) Reset(name string) bool {
	AstLog.Tracef("------- resetting  %s", name)
	for _, vari := range e.variableSnapshotMap {
		if vari.GrlText == name {
			return e.ResetVariable(vari)
		}
	}
	for snap, expr := range e.expressionSnapshotMap {
		if strings.Contains(snap, name) || strings.Contains(expr.GrlText, name) {
			expr.Evaluated = false
		}
	}
	for snap, expr := range e.expressionAtomSnapshotMap {
		if strings.Contains(snap, name) || strings.Contains(expr.GrlText, name) {
			expr.Evaluated = false
		}
	}
	return false
}
```

The function will look for function/variable signature/expression in the working memory. If it founds it, it will mark that
expression's `Evaluate` flag to false so it will be reevaluated again on the next cycle.

**Comment by shreyaspurohit** [comment]
I tested this and it works perfectly. Thank you!
